Remove Python27 support and fix for arrival/departure as datetime objects

eifinger opened the pull request:
If arrival or departure are provided as datetime.datetime objects then urllib will encode it in a format that HERE does not accept.

I implemented a fix which forces the conversion to an ISO8601 string.

Took me 2h to figure this out 😂

abdullahselek commented on herepy/routing_api.py:
```diff
@@ -309,6 +312,13 @@ def matrix(self,
+      """Convert a datetime.datetime object to an ISO8601 string."""
+      if isinstance(datetime_object, datetime.datetime):
+        datetime_object = datetime_object.isoformat()
+      return datetime_object
```
Please leave 4 empty spaces not 2 :)

abdullahselek commented on herepy/routing_api.py:
```diff
@@ -309,6 +312,13 @@ def matrix(self,
+    @staticmethod
+    def _convert_datetime_to_isoformat(datetime_object):
+      """Convert a datetime.datetime object to an ISO8601 string."""
+      if isinstance(datetime_object, datetime.datetime):
```
Leave an empty line after description.

abdullahselek commented on herepy/routing_api.py:
```diff
@@ -309,6 +312,13 @@ def matrix(self,
         return response
 
+    @staticmethod
+    def _convert_datetime_to_isoformat(datetime_object):
```
Please add some unit tests for this method covering date and str parameters.

eifinger commented:
Python 2.7 is failing. I _could_ of course try to provide a fix. But Python 2.7 end of life in 21 days^^

abdullahselek commented:
Yeah we can remove Python 2.7 from CI configurations.

eifinger commented:
I will remove it from the CI files and push another commit.
Should we also remove all the python 27 things like `from __future__`?

eifinger commented:
I removed all remnants of Python 2.7 support I could find.
I also removed the encoding information according to this answer:
https://stackoverflow.com/questions/14083111/should-i-use-encoding-declaration-in-python-3

eifinger commented:
If this is okay for you I would recommend to release a new major version as Python27 support is dropped.

abdullahselek commented:
Changes look good, yes I will release new major version.
